feat: add cart badge

DmytroTarasiuk opened the pull request.

Edyta2801 reviewed (commented):
It would be better if the CartBadge counted the number of items purchased rather than the number of products and CartBadge color should be blue not purple.

DmytroTarasiuk commented:
> It would be better if the CartBadge counted the number of items purchased rather than the number of products and CartBadge color should be blue not purple.

I have changed the code according to your suggestion and now Cart Badge should count purchased items instead of number of single products. In addition, I've also implemented functionality of new user registration from our API. In case, you found any bugs please let me know. Cheers ;)
